### PR TITLE
Rename allowlist to better reflect the wording of modern society

### PR DESCRIPTION
**Pull request by jonathanKingston**
**Reviewer:** @kdzwinel 

## Description:

- In (https://app.asana.com/0/0/1200710125801716/f) the https list was migrated
  - Relevant e-tag code cleaned up on browser load.
- Migrated the old data to new.

## Steps to test this PR:

- Open Firefox
- Go to about:profiles, create new profile, enter "allowlist-migration" as it's name

1. Run "npm run dev-firefox"
2. Launch profile: web-ext run -s build/firefox/dev/ --p allowlist-migration
3. Go to DDG options page
4. "Add unprotected site"
5. Type "example.com" and click Add
6. Now visit "bbc.co.uk"
7. Click "Site Privacy Protection" toggle
8. Click "No"
9. Now visit "cnn.com"
10. Click "Site Privacy Protection" toggle
11. Click "Yes"
12. Send a report for "Something else"
13. Close all tabs besides "about:debugging"
14. Click "Load Temporary Add-on"
15. Checkout the migration branch
16. Check the settings panel lists all three sites
17. Check extension storage object has no legacy keys and they're now renamed to the new key names.
18. Visit example.com and check site toggle status

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/settings.es6.js [hunk 1]
```diff
@@ -31,12 +31,45 @@ function ready () {
     return _ready
 }
 
+// Ensures we have cleared up old storage keys we have renamed
```
Comment (kdzwinel) [inline shared/js/background/settings.es6.js:34]: 👏 Nice! I remember Brad also added clean up function somewhere - should those be consolidated?

File: unit-test/background/storage/https.es6.js [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
 describe('Https storage normal update', () => {
     beforeAll(() => {
         load.loadStub({
             httpsBloom: httpsBloom,
-            httpsWhitelist: httpsWhitelist,
+            httpsAllowlist: httpsAllowlist,
             httpsNegativeBloom: httpsNegativeBloom,
-            httpsNegativeWhitelist: httpsNegativeWhitelist
+            httpsNegativeAllowlist: httpsNegativeAllowlist
```
Comment (kdzwinel) [inline unit-test/background/storage/https.es6.js:14]: should we also use the shorter notation here?

File: unit-test/background/storage/https.es6.js [hunk 3]
```diff
@@ -31,9 +31,9 @@ describe('Https storage bad xhr update', () => {
         badBloom.checksum.sha256 = 'badchecksum'
         load.loadStub({
             httpsBloom: badBloom,
-            httpsWhitelist: httpsWhitelist,
+            httpsAllowlist: httpsAllowlist,
             httpsNegativeBloom: httpsNegativeBloom,
-            httpsNegativeWhitelist: httpsNegativeWhitelist
+            httpsNegativeAllowlist: httpsNegativeAllowlist
```
Comment (kdzwinel) [inline unit-test/background/storage/https.es6.js:36]: ditto (minus the `httpsBloom: badBloom`)

File: shared/js/background/settings.es6.js (hunk 1 above)
Comment (jonathanKingston) [reply shared/js/background/settings.es6.js:35]: @kdzwinel I added "cookieExcludeList" to this list as it should have been removed. I created a follow up https://app.asana.com/0/892838074342800/1200766487104196/f task to clear up other storage usages and document it.

**Comment by kdzwinel** [comment]
I tested allowlisting, migration and all privacy features through the privacy test pages - LGTM!

**Pull request merged**